R/relObsTrends.R

```diff
@@ -20,35 +20,39 @@ relObsTrends<-function(x, timeRes, focalSp){
   yearsAll <- as.numeric(dimnames(x$spatioTemporal)[[2]])
 
   spData<-deconstructOverlay(x$overlaid, attr(x, "visitCol"))
-
-  spData$group<-if(timeRes=="yearly"){
-    spData$year
+  
+  spData$dates<-as.Date(switch(timeRes,
+                     "yearly" = paste0(spData$year, "-01-01"),
+                     "monthly"= paste0(spData$year, "-", sprintf("%02d", spData$month), "-01"),
+                     "daily"  = paste0(spData$year, "-", sprintf("%02d", spData$month), "-", sprintf("%02d", spData$day))))
+
+  res<-if(timeRes=="yearly"){
+    xts::xts(rep(NA, length(seq(min(spData$dates), max(spData$dates), by="year"))), 
+             seq(min(spData$dates), max(spData$dates), by="year"))
   }else if (timeRes=="monthly"){
-    paste0(spData$year,"-",spData$month)
+    xts::xts(rep(NA, length(seq(min(spData$dates), max(spData$dates), by="month"))), 
+             seq(min(spData$dates), max(spData$dates), by="month"))
   }else if(timeRes=="daily"){
-    paste0(spData$year,"-",spData$month,"-",spData$day)
+    xts::xts(rep(NA, length(seq(min(spData$dates), max(spData$dates), by="day"))), 
+             seq(min(spData$dates), max(spData$dates), by="day"))
   }else{
     stop("Unknown time resolution")
   }
```

Comment:
Creates an empty xts with entries for all dates in the time span so that we have NAs for times with no data.